MercurialExtensions/fixutf8/win32helper.py

```diff
@@ -233,15 +233,25 @@ def consolehascp():
 	def rawprint(h, s):
 		InternalWriteFile(h, s)
 
-	def getargs():
+	def getUtf8NonConfigArgs():
 		'''
 		getargs() -> [args]
 
 		Returns an array of utf8 encoded arguments passed on the command line.
+		
+		Skips any --config and following arguments since this is used in a method where
+		those arguments are already removed
 		'''
 		c = INT(0)
 		pargv = CommandLineToArgv(GetCommandLine(), byref(c))
-		return [fromunicode(pargv[i]) for i in xrange(1, c.value)]
+		cleanArguments = []
+		iterator = iter(xrange(1, c.value))
+		for i in iterator:
+			if pargv[i] != "--config":
```

Comment:
I suppose Python lets you get away with that...modifying an iterator within the loop using it strikes me as dangerous.
The other thing that threw me off is that the comment above says "Skips any --config and following arguments" which I read as meaning EVERYTHING after --config, not just the next one.